### DISCUSSION TITLE
Models that support Precipitation Probability?

### DISCUSSION THREAD

**Discussion by petergault** [open-meteo/open-meteo#934] [General]
Before I jump in - Patrick, Open Moteo is awesome. It's one of the coolest things I've ever seen on the internet. That you have assembled all of these datasets together and provided them for free for non-commercial use is incredible. This is some impressive. I found Open Meteo after the New York Times article on GraphCast and it's awesome that you got the model up and running. 

For context, I am building an app that only focuses on "when will it rain?" that utilizes different weather APIs to provide a very, simple clean way of seeing hourly rain predictions for the next 7 days (168 hours). 

Poking around in different models now, Best Match and GFS Seamless provide rain probability predictions, but the other services don't. 
<img width="992" alt="Screenshot 2024-08-02 at 1 23 36 AM" src="https://github.com/user-attachments/assets/06a1291c-4546-4f92-ae87-1b81469b67ed">

Looking at that this card, it seems like this was solved in April, so I am not sure if this is a bug or if these models don't support prediction:
https://github.com/open-meteo/open-meteo/discussions/708

One suggestion as well - it would be great if each model can have a tooltip that explains more about the model and how it differs from the other options. You have the explanations in the model links but there a bunch of models without those info boxes.

**Comment by patrick-zippenfenig** [comment]
Hi, thanks for asking.

Precipitation probabilities are calculated from ensemble models. Only a few ensemble models are available.

<img width="1109" alt="Screenshot 2024-08-06 at 15 12 25" src="https://github.com/user-attachments/assets/960af719-7700-47ba-9ed8-b0c63f35cb26">

Note: GEM is supposed to generate precipitation probabilities, but apparently there is a bug...

**Reply by patrick-zippenfenig** [reply]
Fix for CMC GEM models will be deployed soon: https://github.com/open-meteo/open-meteo/pull/979

EDIT: GEM precipitation probability is working again

**Comment by guidocioni** [comment]
Hey @patrick-zippenfenig regarding this, could you clarify how the `best_match` precipitation probability is computed? 

![Screenshot 2024-08-20 at 17 23 56](https://github.com/user-attachments/assets/7b67c1e6-7378-44d9-b0bc-cde137c044b7)

it seems there is no match between the different models for which the probability is computed and the `best_match` option. If anything it is close to `gfs_seamless` but for this location even the precipitation probability should be the one from `icon_seamless`.

**Reply by patrick-zippenfenig** [reply]
It should use the corresponding probabilities for each model. ICON-EPS for ICON. There could be a bug that an older version of precipitation probabilities are still used which are based on GFS but use a slightly different calculation. I will create a ticket and check it next week

**Reply by patrick-zippenfenig** [reply]
The interfering old code has now been removed: https://github.com/open-meteo/open-meteo/pull/977. The API is now selecting the correct precipitation probability from the correct ensemble domain. [ICON EPS+EU in this example](https://open-meteo.com/en/docs#current=&hourly=precipitation_probability&daily=&start_date=2024-08-10&end_date=2024-08-15&time_mode=time_interval&models=best_match,ecmwf_ifs025,bom_access_global,gfs_seamless,icon_seamless,gem_seamless)
